database/seeders/GameReleasesSeeder.php

```diff
@@ -0,0 +1,155 @@
+                case 45: // Intellivision
+                    $range = [1979, 1990];
+                    break;
+                case 7: // NES
+                    $range = [1983, 1995];
+                    break;
+                case 1: // MegaDrive
+                    $range = [1988, 1997];
+                    break;
+                case 4: // GameBoy
+                    $range = [1989, 2003];
+                    break;
+                case 15: // GameGear
+                    $range = [1990, 1997];
+                    break;
+                case 3: // SNES
+                    $range = [1990, 1999];
+                    break;
+                case 12: // PSX
+                    $range = [1994, 2006];
+                    break;
+                case 2: // N64
+                    $range = [1996, 2002];
+                    break;
+                case 6: // GameBoy Color
+                    $range = [1998, 2003];
+                    break;
+                case 21: // PS2
+                    $range = [2000, 2013];
+                    break;
+                case 16: // GameCube
+                    $range = [2001, 2007];
+                    break;
+                case 5: // GameBoy Advance
+                    $range = [2001, 2010];
+                    break;
+                case 41: // PSP
+                    $range = [2004, 2014];
+                    break;
+                default:
+                    // unhandled. weight towards the newest date
+                    $year = (int) sqrt(random_int(1976 * 1976, 2012 * 2012));
+                    break;
+            }
+            if ($year === 0) {
+                // weight ranged releases towards the initial date
+                $year = $range[1] - (int) sqrt(random_int(0, pow($range[1] - $range[0] + 1, 2)));
+            }
+
+            $release->released_at = Carbon::createFromDate($year, 1, 1);
+            if ($year < 1985 && random_int(0, (1986 - $year) * (1987 - $year)) > 5) {
+                // 17% (1/6) chance of only being year for 1984
+                // 75% (15/20) chance of only being year for 1982
+                // 91% (51/56) chance of only being year for 1980
+                // 95% (105/110) chance of only being year for 1976
+                $release->released_at_granularity = ReleasedAtGranularity::Year;
+            } elseif ($year < 1988 && random_int(0, (1990 - $year) * (1991 - $year)) > 9) {
+                // 25% (3/12) chance of month granularity for 1987
+                // 70% (21/30) chance of month granularity for 1985
+                // 87% (61/70) chance of month granularity for 1982 (after 25% chance of it not being year only)
+                $release->released_at = $release->released_at->addMonths(rand(0, 11));
+                $release->released_at_granularity = ReleasedAtGranularity::Month;
+            } else {
+                $release->released_at = $release->released_at->addDays(rand(0, 364));
```

Comment:
Consistency: some places use `rand()`, others use `random_int()`. I'm not sure if their behavior is different or if it's worth conforming the whole seeder to use just one.